### PR TITLE
Proposed ECMP Next Hop CLI Show Cmd Output Format

### PR DESCRIPTION
**Pull request by kktheballer**
Adding CLI command formats for

show fgnhg hash-view <fg-nhg-group-name>
show fgnhg active-hops <fg-nhg-group-name>

### CODE REVIEW

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 1]
```diff
@@ -189,6 +189,29 @@ CLI commands:
 	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
 
+### 2.3.1 CLI 'show fg nhg hash-view' Output Format
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:192]: This supports a fg-nhg-group-name argument as listed in above section?

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 2]
```diff
@@ -189,6 +189,29 @@ CLI commands:
 	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
 
+### 2.3.1 CLI 'show fg nhg hash-view' Output Format
+```
+Next-Hop-Key    Hash Bucket #
+--------------  -----------------------------------------------------------------------------------------------------------------------
+200.200.200.1   ['88', '89', '102', '103', '100', '101', '99', '98', '91', '90', '93', '92', '95', '94', '97', '96']
+200.200.200.2   ['41', '20', '21', '22', '23', '15', '17', '16', '19', '18', '46', '44', '45', '42', '43', '40']
+200.200.200.3   ['64', '65', '66', '67', '68', '69', '77', '76', '75', '74', '73', '72', '71', '70', '78']
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:198]: How about 88, 89, 102, 103 without the array like syntax and without single quotes?

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 3]
```diff
@@ -189,6 +189,29 @@ CLI commands:
 	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
 
+### 2.3.1 CLI 'show fg nhg hash-view' Output Format
+```
+Next-Hop-Key    Hash Bucket #
+--------------  -----------------------------------------------------------------------------------------------------------------------
+200.200.200.1   ['88', '89', '102', '103', '100', '101', '99', '98', '91', '90', '93', '92', '95', '94', '97', '96']
+200.200.200.2   ['41', '20', '21', '22', '23', '15', '17', '16', '19', '18', '46', '44', '45', '42', '43', '40']
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:197]: A sorted set of integer values for 'Hash Bucket #' may be better

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 4]
```diff
@@ -189,6 +189,29 @@ CLI commands:
 	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:190]: Can we also add details about cli output for 'show fg nhg group'?

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 5]
```diff
@@ -185,11 +185,114 @@ CLI commands:
...
+```
+ 
+NOTE: <fg-nhg-group-name> is an optional parameter containing the user-defined alias of nhg ipv4 or nhg ipv6 
+found in 'FG_NHG_PREFIX' section of config dB. If specified, the output will display either ipv4 next hops and banks
+or ipv6 next hops and banks accordingly. If it is not specified, by default both ipv4 and ipv6 next hops and banks 
+are displayed as shown below:
+
+
++-----------------------------+--------------------+------------+
+| FG_NHG_PREFIX               | Active Next Hops   | Bank Ids   |
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:202]: Recommend changing ```Bank Ids``` to ```Hash buckets``` and NIT: ```Active Next Hops``` to ```Next Hop```

Comment (kktheballer) [reply doc/ecmp/fine_grained_next_hop_hld.md:202]: Ok will adjust both PRs accordingly

**Review by anish-n — APPROVED** [review]
Looks good

**Review by abdosi — APPROVED** [review]
LGTM

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 6]
```diff
@@ -1,373 +1,477 @@
...
-- Change the DUT route entry to have all next-hops in a bank0 as down, make sure that the traffic now flows to links in bank1 only
-- Change the DUT route entry to add 1st next-hop in a previously down bank0, now some of the flows should migrate to the newly added next-hop
-- Disable a link from the link mapping created in FG_NHG_MEMBER and validate that flows were redistributed in the same bank and occured in a consistent fashion
-- Validate that in all cases the flow distribution per next-hop is roughly equal
-- Test both IPv4 and IPv6 above
-- The above test is configured via config_db entries directly, a further test mode to configure Fine Grained ECMP via minigraph will be present and tested
-- Test warm reboot to ensure there is no traffic disruption and ECMP groups are correctly applied post warm boot
+﻿# SONiC Fine Grained ECMP 
+# High Level Design Document
+### Rev 1.2
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:3]: Please can we update the rev number here to the latest 1.3.

**Comment by anish-n** [comment]
Why does it appear that the whole file was modified, hope we can correct that since it helps with maintaining version history.

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 7]
```diff
@@ -191,11 +192,113 @@ CLI commands:
...
-	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
+	show fgnhg hash-view <fg-nhg-group-name> (shows the current hash bucket view of fg nhg)
+	show fgnhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
 
-Show CLI commands of ```show fg nhg hash-view``` and ```show fg nhg active-hops``` are implemented as a view of the state db table described in section 2.2
+### 2.3.1 CLI 'show fgnhg hash-view <fg-nhg-group-name>' Output Format
+```
+ 
+NOTE: <fg-nhg-group-name> is an optional parameter containing the user-defined alias of nhg ipv4 or nhg ipv6
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:202]: Suggest that we reword this to say ```NOTE: <fg-nhg-group-name> is an optional parameter containing the user-defined alias defined via FG_NHG entry ```

File: doc/ecmp/fine_grained_next_hop_hld.md [hunk 8]
```diff
@@ -191,11 +192,113 @@ CLI commands:
...
 	config fg nhg member <add/del> <fg-nhg-group-name>  <next-hop-ip>
 	show fg nhg group <fg-nhg-group-name/all>
-	show fg nhg hash-view <fg-nhg-group-name> (shows the current hash bucket view of fg nhg)
-	show fg nhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
+	show fgnhg hash-view <fg-nhg-group-name> (shows the current hash bucket view of fg nhg)
+	show fgnhg active-hops <fg-nhg-group-name> (shows which set of next-hops are active) 
 ```
 
-Show CLI commands of ```show fg nhg hash-view``` and ```show fg nhg active-hops``` are implemented as a view of the state db table described in section 2.2
+### 2.3.1 CLI 'show fgnhg hash-view <fg-nhg-group-name>' Output Format
```
Comment (anish-n) [inline doc/ecmp/fine_grained_next_hop_hld.md:199]: image
This does not show up as expected when viewed as markdown in the reader, maybe a markdown syntax issue, can you please fix the same

Comment (kktheballer) [reply doc/ecmp/fine_grained_next_hop_hld.md]: Addressed and made some more changes within section description

**Comment by abdosi** [comment]
please add details in description
